src/data-interfaces/HoloFuelDnaInterface.js

```diff
@@ -10,6 +10,25 @@ const createZomeCall = instanceCreateZomeCall(INSTANCE_ID)
 
 const MOCK_DEADLINE = '4019-01-02T03:04:05.678901234+00:00'
 
+// Creates an array of all counterparties for a provided transaction list
+export async function getTxCounterparties (transactionList) {
+  const counterpartyList = transactionList.map(tx => {
+    const { counterparty } = tx
+    return counterparty
+  })
+  const agentDetailsList = []
+  for (let i = 0; i < counterpartyList.length; i++) {
+    const agentId = counterpartyList[i]
+    const agent = await new Promise(resolve => {
+      const findAgent = HoloFuelDnaInterface.user.getCounterparty({ agentId })
+      resolve(findAgent)
+    })
```

Comment:
I believe this can be simplified to
```
const agent = await HoloFuelDnaInterface.user.getCounterparty({ agentId })
```

Reply:
but there's a function in utils that makes this even simpler. I think you can replace lines 19-27 with

```
import { promiseMap } from 'utils'

...

const agentDetailsList = await promiseMap(counterpartyList, agentId => HoloFuelDnaInterface.user.getCounterparty({ agentId }))
```

Reply:
So, at first, I did actually write `const agent = await HoloFuelDnaInterface.user.getCounterparty({ agentId })`.  

However, because we're calling await within an iterator and the return value of async functions is always a Promise, `agentDetailsList` will return an Array of unresolved promises. 
 Consequently, the value for each iteration must be resolved before we can make use of the resulting Array, which is why each all to `HoloFuelDnaInterface.user.getCounterparty({ agentId })` is wrapped in a Promise.

Reply:
Another option is to wrap a map function of the `counterpartyList` array in a Promise.all(); however, from my experience, this approach requires a bit more setup / code to attain the same outcome.

Reply:
Open to new ideas or additional thoughts though. :+1:

Reply:
Got it. Just read through `promiseMap`, and it will do the trick.  Imported and implemented. :+1: